### PR TITLE
M2P-161 Stop pushing model/quote JS script

### PR DESCRIPTION
**Pull request by vitaliyreznikov**
We introduced an issue in https://github.com/BoltApp/bolt-magento2/pull/796. Actually it was known magento issue that was fixed in new Magento versions, see https://github.com/magento/magento2/pull/23099 , https://github.com/magento/magento2/issues/14412, https://github.com/magento/magento2/pull/21432

Under some conditions, Magento JS script produced an error because they initialated wrong.

We had a fix https://github.com/BoltApp/bolt-magento2/pull/808 but now I see it doesn't affect the root of the issue.

In PR 796 we added `require(['Magento_Checkout/js/model/quote'] ...)` thing. This requireJS API means we push `model/quote` script to load if it wasn't loaded before. Under some very rare conditions, it triggers incorrect initializing.

So solution the same as in https://github.com/magento/magento2/pull/21432 : don't load module/quote script before DOM.loaded. We know that when DOM.loaded we have all parameters to initialize this script.

The issue about a race condition, so it's quite hard to reproduce it. Looks like it happens very rarely.
We need some (or all) of the following conditions:
- site works slowly
- browser work slowly (IE11 for example)
- the first page load, so we don't have scripts in browser cache.
- site works in developer mode

Fixes: [M2P-161]

#changelog M2P-161 Stop pushing model/quote JS script

# Type of change

- [x] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [x] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

File: view/frontend/templates/js/replacejs.phtml [hunk 1]
```diff
@@ -2321,12 +2321,12 @@ if (!$block->isSaveCartInSections()) { ?>
     });
 
     if (trim(location.pathname, '/') === 'checkout/cart') {
+        require(['domReady!'], function() {
         require([
         'jquery',
-        'Magento_Customer/js/customer-data',
         'Magento_Checkout/js/model/quote',
         'domReady!'
```
Comment (matt-thomason) [inline view/frontend/templates/js/replacejs.phtml:2328]: This means we do not need this second `domReady!`, right?

File: view/frontend/templates/js/replacejs.phtml [hunk 2]
```diff
@@ -2321,12 +2321,12 @@ if (!$block->isSaveCartInSections()) { ?>
     });
 
     if (trim(location.pathname, '/') === 'checkout/cart') {
+        require(['domReady!'], function() {
         require([
```
Comment (matt-thomason) [inline view/frontend/templates/js/replacejs.phtml:2325]: Can you fix indentation?

**Comment by vitaliyreznikov** [comment]
> What about the other two instances in this replacejs.phtml file where we rely on domReady!?

Good point, but other things are work :) and we haven't changed them for months. Looks like only '`Magento_Checkout/js/model/quote'` is brittle enough and we use it only in place I fix in this PR.

Maybe it makes sense to run all code after domReady, but I think it's too big change for hotfix PR. Let's discuss this during other JS refactoring tasks..

File: view/frontend/templates/js/replacejs.phtml (hunk 1 above)
Comment (vitaliyreznikov) [reply view/frontend/templates/js/replacejs.phtml]: Correct, thank you

File: view/frontend/templates/js/replacejs.phtml (hunk 2 above)
Comment (vitaliyreznikov) [reply view/frontend/templates/js/replacejs.phtml]: Done

**Comment by matt-thomason** [comment]
That is reasonable, thanks for talking it out.

**Pull request merged**